File: frontend/types/RoomTicket.ts
export interface RoomTicket {
  id: string;
  customer_id: string;
  room_id: string;
  check_in_date: string;
  check_out_date: string;
  number_of_occupants: number;
  status: number;
  created_at: string;
}

export interface CreateRoomTicketPayload {
  customer_id: string;
  room_id: string;
  number_of_occupants: number;
}

export interface UpdateRoomTicketPayload {
  customer_id?: string;
  room_id?: string;
  number_of_occupants?: number;
}

File: server/Models/RoomTicket.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Yes.Models;

public class RoomTicket
{
    [Key]
    public string Id { get; set; } = string.Empty;

    [Required]
    public string Customer_id { get; set; } = string.Empty;
    [ForeignKey("Customer_id")]
    public virtual Customer Customer { get; set; } = null!;
    [Required]
    public string Room_id { get; set; } = string.Empty;
    [ForeignKey("Room_id")]
    public virtual Room Room { get; set; } = null!;

    public DateTime? CheckInDate { get; set; } = null;
    public DateTime? CheckOutDate { get; set; } = null;
    [Required]
    public int Number_of_occupants { get; set; } = 1;
    [Required]
    public int Status { get; set; } = 0;
    [Required]
    public DateTime Created_at { get; set; } = DateTime.Now;
}
